Community/Tdarr_Plugin_goof1_URL_Plex_Refresh.js

```diff
@@ -0,0 +1,221 @@
+
+              *Note* If this number is wrong everything will behave as though it's working great but the folder will simply not be scanned. 
+
+              https://support.plex.tv/articles/201638786-plex-media-server-url-commands/
+
+               \\nExample:\\n
+               29`,
+    },
+    {
+      name: 'Plex_Path',
+      type: 'string',
+      defaultValue: '',
+      inputUI: {
+        type: 'text',
+      },
+      tooltip: `
+              If the Plex path is not the same as the local path you may need to sub parts of the path. 
+              Here is where you would enter the path that Plex uses to find the file. You would only enter the part of the path that is different.
+
+              If your TDarr path is:
+              /media/local/tv/The Best Show Evaaaarr/Season 2/The Best Show Evaaaarr - S02E31 - Heck Yea HDTV-720p.mp4
+              
+              And the Plex path to the file is:
+              /data/tv/The Best Show Evaaaarr/Season 2/The Best Show Evaaaarr - S02E31 - Heck Yea HDTV-720p.mp4
+
+              then part you would enter here is:
+
+               \\nExample:\\n
+               /data/`,
+    },
+    {
+      name: 'TDarr_Path',
+      type: 'string',
+      defaultValue: '',
+      inputUI: {
+        type: 'text',
+      },
+      tooltip: `
+              If the Plex path is not the same as the local path you may need to sub parts of the path. 
+              Here is where you would enter the path that Plex uses to find the file. You would only enter the part of the path that is different.
+
+              If your TDarr path is:
+              /media/local/tv/The Best Show Evaaaarr/Season 2/The Best Show Evaaaarr - S02E31 - Heck Yea HDTV-720p.mp4
+              
+              And the Plex path to the file is:
+              /data/tv/The Best Show Evaaaarr/Season 2/The Best Show Evaaaarr - S02E31 - Heck Yea HDTV-720p.mp4
+
+              then part you would enter here is:
+
+               \\nExample:\\n
+               /media/local/`,
+    },
+  ],
+});
+
+const plugin = (file, librarySettings, inputs, otherArguments) => {
+  inputs = loadDefaultValues(inputs, details);
+
+  let response = {
+    file,
+    removeFromDB: false,
+    updateDB: false,
+    infoLog: '',
+  };
+
+  const http = require('http')
+  const https = require('https')
+
+  const type = inputs.Url_Protocol
+  const url = inputs.Plex_Url
+  const port = inputs.Plex_Port
+  const token = inputs.Plex_Token
+  const key = inputs.Library_Key
+  const plexPath = inputs.Plex_Path
+  const tdarrPath = inputs.TDarr_Path
+
+  if (!type || !url || !token || !key) {
+    response.infoLog = `Url_Protocol, Plex_Url, Plex_Token, and Library_Key are all required`
+    response.processFile = false
+    return response
+  }
+
+
+  let filePath = file.file
+  filePath = filePath.substring(0, filePath.lastIndexOf("/"));
+
+  if (tdarrPath || plexPath) {
+    filePath = filePath.replace(tdarrPath, plexPath)
+  }
```

Comment:
Seems to me that this check should be `&&` and not `||`

Reply:
it's possible that the root of one mapping is not the root of the other. the replace with then just remove the specified portion of the path.

Reply:
Yeah, but that's what I'm implying. The if statement checks if either isn't empty or null, thus now it could happen that someone only fills `tdarrPath` and forgets about `plexPath` (somehow) or the other way around. It'll replace `/data/something/` with an empty string, thus removing that part entirely. The other way around, it would replace nothing with something.

Reply:
and that could also be the case. if plex is /tv and tdarr is /data/media/tv you would need to remove /data/media for the scan so plex would be plank and /data/media would be in tdarr. If tdarr is /tv and plex is /data/media/tv you would need to type / in tdarr and /data/media in plex to make the right string. i might be able to do some complex regex or make it a few else if's that series here to handle all the possible combos and some edge cases. Not sure if it'd be worth it though. Think its worth trying?

Reply:
got bored and played for a bit. i think this fulfills your concerns with having the tdarrPath be blank but having something in plex that needs to be added. Thoughts?

```javascript
  if (tdarrPath && !plexPath) {
    filePath = filePath.replace(tdarrPath, plexPath);
  } else if (!tdarrPath && plexPath) {
    filePath = filePath.replace(/^/, plexPath);
  }
```